Mathlib/Algebra/Category/ModuleCat/ChangeOfRings.lean

```diff
@@ -120,7 +120,7 @@ theorem restrictScalars.smul_def' {R : Type u₁} {S : Type u₂} [Ring R] [Ring
 
 instance (priority := 100) sMulCommClass_mk {R : Type u₁} {S : Type u₂} [Ring R] [CommRing S]
     (f : R →+* S) (M : Type v) [I : AddCommGroup M] [Module S M] :
-    have : SMul R M := (RestrictScalars.obj' f (ModuleCat.mk M)).isModule.toSMul
+    haveI : SMul R M := (RestrictScalars.obj' f (ModuleCat.mk M)).isModule.toSMul
```

Comment:
Could you explain why `haveI` instead of `have`? I just want to make sure you're not using it with the Lean 3 meaning. Now it means "inlined have", where `(RestrictScalars.obj' f (ModuleCat.mk M)).isModule.toSMul` itself will be substituted into wherever `this` is used.

Reply:
If I understand the commit correctly, the inlining is the point, since otherwise the declaration will have head symbol `have` instead of `SMulCommClass`.

Reply:
Ah, thanks, I didn't notice this was for the type of the instance.

Reply:
to be honest I'm still not sure how useful this instance will will be with such a `haveI`ed subterm, but at least this stops it breaking doc-gen